src/inquirer.ts

```diff
@@ -0,0 +1,127 @@
+
+interface DeploymentIdPrompt {
+  name: string;
+  value: script_v1.Schema$Deployment;
+}
+
+/**
+ * Inquirer prompt for a deployment Id.
+ * @param {DeploymentIdPrompt[]} choices An array of `DeploymentIdPrompt` objects.
+ * @returns {Promise<{ deploymentId: string }>} A promise for an object with the `deploymentId` property.
+ */
+export const deploymentIdPrompt = (choices: DeploymentIdPrompt[]) => prompt<{ deploymentId: string }>([
+  {
+    choices,
+    message: 'Open which deployment?',
+    name: 'deploymentId',
+    type: 'list',
+  },
+]);
+
+/**
+ * Inquirer prompt for a project description.
+ * @returns {Promise<{ description: string }>} A promise for an object with the `description` property.
+ */
+export const descriptionPrompt = () => prompt<{ description: string }>([{
+  default: '',
+  message: LOG.GIVE_DESCRIPTION,
+  name: 'description',
+  type: 'input',
+}]);
+
+export interface PromptAnswers {
+  doAuth: boolean; // in sync with prompt
+  localhost: boolean; // in sync with prompt
+}
+
+/**
+ * Inquirer prompt for oauth scopes.
+ * @returns {Promise<PromptAnswers>} A promise for an object with the `PromptAnswers` interface.
+ */
+export const oauthScopesPrompt = () => prompt<PromptAnswers>([{
+  message: 'Authorize new scopes?',
+  name: 'doAuth',
+  type: 'confirm',
+}, {
+  message: 'Use localhost?',
+  name: 'localhost',
+  type: 'confirm',
+  when: (answers: PromptAnswers) => {
+    return answers.doAuth;
+  },
+}]);
+
+/**
+ * Inquirer prompt for overwriting a manifest.
+ * @returns {Promise<{ overwrite: boolean }>} A promise for an object with the `overwrite` property.
+ */
+export const overwritePrompt = () => prompt<{ overwrite: boolean }>([{
+  default: false,
+  message: 'Manifest file has been updated. Do you want to push and overwrite?',
+  name: 'overwrite',
+  type: 'confirm',
+}]);
+
+/**
+ * Inquirer prompt for project Id.
+ * @returns {Promise<{ projectId: string }>} A promise for an object with the `projectId` property.
+ */
+export const projectIdPrompt = () => prompt<{ projectId: string }>([{
+  message: `${LOG.ASK_PROJECT_ID}`,
+  name: 'projectId',
+  type: 'input',
+}]);
+
+export interface ScriptIdPrompt {
+  name: string;
+  value: string;
+}
+
+/**
+ * Inquirer prompt for script Id.
+ * @param {ScriptIdPrompt[]} fileIds An array of `ScriptIdPrompt` objects.
+ * @returns {Promise<{scriptId: string;}>} A promise for an object with the `scriptId` property.
+ */
+export const scriptIdPrompt = (fileIds: ScriptIdPrompt[]) => prompt<{ scriptId: string }>([
+  {
+    choices: fileIds,
+    message: LOG.CLONE_SCRIPT_QUESTION,
+    name: 'scriptId',
+    pageSize: 30,
+    type: 'list',
+  },
+]);
+
+/**
+ * Inquirer prompt for script type.
+ * @returns {Promise<{ type: string }>} A promise for an object with the `type` property.
+ */
+export const scriptTypePrompt = () => prompt<{ type: string }>([{
+  choices: Object.keys(SCRIPT_TYPES).map((key) => SCRIPT_TYPES[key as any]), // tslint:disable-line: no-any
```

Comment:
Try: `(key: string)`

No `key as any`. (no-any)
Maybe `key as string` if required.

Reply:
The correct cast to access elements of an enum is the ugly `[key as unknown as number]`